arch/x86/core/early_serial.c

```diff
@@ -69,6 +72,11 @@ static void serout(int c)
 
 int arch_printk_char_out(int c)
 {
+	if (!early_serial_init_done) {
+		suppressed_chars++;
+		return c;
```

Comment:
Can we improve this by using only a single return statement in this function? It would help us comply with coding guideline on this topic.

Reply:
I don't think it results in safer/better code to follow such a guideline blindly. It makes sense when you have a clear cleanup block of code that needs to always run before returning from the function, but otherwise its benefit is questionable (IMO). I took a quick look at `arch/x86/core/*.c` and there are lots of places which violate such a guideline. If we really were to apply it with no exceptions I think it might make more sense to do that in a single sweep of the x86 architecture code, rather than introducing internal style inconsistencies.